crates/datasources/Cargo.toml

```diff
@@ -13,6 +13,8 @@ apache-avro = "0.16"
 async-channel = "2.1.1"
 async-stream = "0.3.5"
 async-trait = { workspace = true }
+arrow-array = { version = "50.0.0", default-features = false }
+arrow-schema = { version = "50.0.0", default-features = false }
```

Comment:
use the version of arrow that's re-exported via the datafusion package `datafusion::arrow`

Reply:
Yeah I'm having some trouble using the `downcast_dictionary_array!` macro in write/demux without adding that crate explicitly 🤔 . I'm far from an expert on Rust macros so maybe I'm missing something obvious?

Reply:
Have removed since this isn't needed unless we want to support dictionaries.